src/transforms/pipelines/config.rs

```diff
@@ -42,62 +52,178 @@ impl Clone for PipelineConfig {
     }
 }
 
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipeline")]
+impl TransformConfig for PipelineConfig {
+    async fn build(&self, ctx: &TransformContext) -> crate::Result<Transform> {
+        let condition = match &self.filter {
+            Some(config) => Some(config.build(&ctx.enrichment_tables)?),
+            None => None,
+        };
+
+        // Setup the interior transform chain in this pipeline.
+        //
+        // NOTE: As this pipeline is not expanding to separate transforms we
+        // lose validation that happens at the topology layer (e.g. making sure
+        // input/output types of connected transforms are compatible). This is
+        // an unfortunate consequence of having a "topology in a topology".
+        //
+        // In the future, it would be great to decouple that type of validation
+        // from the physical task boundaries. You could imagine a future where
+        // we replace transform expansion with the ability to return different
+        // types of compound transforms directly from TransformConfig::build,
+        // and those know how to play nice with a subsequent graph-based
+        // validation scheme.
+        //
+        // For now, considering that most everything is log to log with regard
+        // to pipelines we avoid adding duplicate validation here in favor of
+        // future work.
+        if self.transforms.is_empty() {
+            return Err(format!("empty pipeline: {}", self.name).into());
+        }
```

Comment:
This is why I asked if it was possible for there to be an empty transform set below.

Reply:
Ah, right, right. I forgot that I ended up disabling one of the soak tests over this. @binarylogic product question, is https://github.com/vectordotdev/vector/blob/a3c0d26a41bbce4765dda08f7566d3d09f680a97/soaks/disabled-tests/http_datadog_filter_blackhole/vector.toml a valid config? That is, do we intend to support pipelines that have no transforms associated with them?

Reply:
I see. The previous code here allowed for zero-transform pipelines maybe accidentally? The new PR does not allow for zero-transform pipelines, considering them as a configuration error. If we _do_ want to allow for zero-transform pipelines going forward should we emit a warning -- we did, in fact, have a silent bug in the http->pipelines config where transforms were accidentally not hooked up to a pipeline -- or what? We could also choose to elide them from the filtering series, since they'll do no work they will only add overhead, I think.

Reply:
Yes, we can prohibit zero transform pipelines. There is no reason to have a pipeline without transforms.